containers/Map.hpp

```diff
@@ -158,8 +163,31 @@ namespace ft
 				return const_iterator(btree_end(_root));
 			}
 
+			void erase (iterator position)
+			{
+				value_type *pair_erased = btree_delete<value_type>(&_root, *position, &_compare_value_type<value_type, Compare>);
+				if (pair_erased)
+					this->_allocator.deallocate(pair_erased, 1);
+			}
+
+			size_type erase (const key_type& key)
+			{
+				value_type *pair_erased = btree_delete<value_type>(&_root, ft::make_pair(key, mapped_type()), &_compare_value_type<value_type, Compare>);
+				if (!pair_erased)
+					return (0);
+				this->_allocator.deallocate(pair_erased, 1);
+				return (1);
+			}
+
+			void erase (iterator first, iterator last)
+			{
+				first--;
+				while (first != last)
+					this->erase(first++);
```

Comment:
tem um problema aqui quando mandam o begin e ele tenta apagar um pair que não existe
tentei várias coisas pra solucionar, por exemplo
```suggestion
				first--;
				while (first != last)
				        if (*first)
					    this->erase(first++);
```
mas nada ele me deixava compilar
resolvi abrir o PR e vermos se temos alguma luz pra esse caso do begin
achei a solução tão elegante de resto 🥲